src/widget/form/profileform.cpp

```diff
@@ -247,10 +247,14 @@ void ProfileForm::onRenameClicked()
     Nexus& nexus = Nexus::getInstance();
     QString cur = nexus.getProfile()->getName();
     QString title = tr("Rename \"%1\"", "renaming a profile").arg(cur);
+
     do
     {
-        QString name = QInputDialog::getText(this, title, title+":");
-        if (name.isEmpty()) break;
+        QString name = QInputDialog::getText(0, title, title + ":");
```

Comment:
Sorry, my mistake. It's a widget attribute, not a window flag. Hm ... only way I see is to implement it right (not use the convenience method):

```c++
    QInputDialog dialog(this);
    // probably this one:
    dialog.setAttribute(Qt::WA_WindowPropagation, true);

    dialog.setWindowTitle(title);
    dialog.setLabelText(title + ":");
    int ret = dialog.exec();

    QString name;
    if (ret) {
        name = dialog.textValue();
    }
}
```

Does this help?